what4/src/What4/Expr.hs

```diff
@@ -97,3 +107,9 @@ import What4.Expr.Builder
 import What4.Expr.GroundEval
 import What4.Expr.WeightedSum
 import What4.Expr.UnaryBV
+
+
+-- | A \"dummy\" data type that can be used for the
+--   user state field of an @ExprBuilder@ when there
+--   is no other interesting state to track.
+data EmptyExprBuilderState t = EmptyExprBuilderState
```

Comment:
Now that we have `EmptyExprBuilderState`, we should use it throughout the `what4` packages consistently. I spot the following places where it could be used:

* https://github.com/GaloisInc/what4/blob/aad3290b315b0bb7fd49e0e95729133ae24f4863/what4-transition-system/test/Main.hs#L65-L66
* https://github.com/GaloisInc/what4/blob/aad3290b315b0bb7fd49e0e95729133ae24f4863/what4/doc/QuickStart.hs#L24
* https://github.com/GaloisInc/what4/blob/aad3290b315b0bb7fd49e0e95729133ae24f4863/what4/test/AdapterTest.hs#L42
* https://github.com/GaloisInc/what4/blob/aad3290b315b0bb7fd49e0e95729133ae24f4863/what4/test/ExprBuilderSMTLib2.hs#L55
* https://github.com/GaloisInc/what4/blob/aad3290b315b0bb7fd49e0e95729133ae24f4863/what4/test/ExprsTest.hs#L37
* https://github.com/GaloisInc/what4/blob/aad3290b315b0bb7fd49e0e95729133ae24f4863/what4/test/IteExprs.hs#L39
* https://github.com/GaloisInc/what4/blob/aad3290b315b0bb7fd49e0e95729133ae24f4863/what4/test/OnlineSolverTest.hs#L53
* https://github.com/GaloisInc/what4/blob/aad3290b315b0bb7fd49e0e95729133ae24f4863/what4/test/TestTemplate.hs#L56
* https://github.com/GaloisInc/what4/blob/aad3290b315b0bb7fd49e0e95729133ae24f4863/what4/README.md?plain=1#L78

Reply:
yeah, probably a good idea